lib/Client.js

```diff
@@ -2173,6 +2173,16 @@ class Client extends EventEmitter {
             activities = [activities];
         }
         if(activities !== undefined) {
+            if(activities.length > 0) {
+                activities = activities.map((a) => {
+                    if(a.hasOwnProperty("type")) {
+                        a.type = a.url ? 1 : 0;
+                    } else if(a.type === 4) {
+                        a.state = a.name;
+                    }
```

Comment:
Code here will always result in the type being set to 0 or 1 if a type is provided, even if it is `4`. Would rather see a state prop to be in line with Discord